site/app/controllers/student/SubmissionController.php

```diff
@@ -491,13 +491,17 @@ public function ajaxBulkUpload($gradeable_id) {
                 // exec() and similar functions are disabled by security policy,
                 // so we are using a python script via CGI to validate whether file is divisible by num_page or not.
                 $pdf_full_path = FileUtils::joinPaths($pdf_path, $job_data["timestamp"], $job_data["filename"]);
+                $num_pages_encoded = urlencode($num_pages);
+                $file_name_encoded = urlencode($job_data['filename']);
+                $pdf_full_path_encoded = urlencode($pdf_full_path);
                 $ch = curl_init();
-                curl_setopt($ch, CURLOPT_URL, $this->core->getConfig()->getCgiUrl() . "pdf_page_check.cgi?pdf_path={$pdf_full_path}&num_page={$num_pages}&file_name={$job_data['filename']}");
+                curl_setopt($ch, CURLOPT_URL, $this->core->getConfig()->getCgiUrl() . "pdf_page_check.cgi?pdf_path={$pdf_full_path_encoded}&num_page={$num_pages_encoded}&file_name={$file_name_encoded}");
```

Comment:
Sorry to make you change it again but could you actually swap this to use `http_build_query()`. It'll make this much cleaner and more maintainable. A good existing example of this would be: https://github.com/Submitty/Submitty/blob/de454f23ee1a2828f62c89f717c4043be969696d/site/app/controllers/HomePageController.php#L194-L201

Reply:
Don't change the `curl_setopt` to `curlRequest` just use the example for how to use `http_build_query`